lib/Parse/ParseStmt.cpp

```diff
@@ -1833,6 +1833,67 @@ Parser::parseStmtConditionElement(SmallVectorImpl<StmtConditionElement> &result,
   return Status;
 }
 
+/// Returns `true` if the current token represents the start of a conditional statement body.
+bool Parser::isStartOfConditionalStmtBody() {
+  Parser::BacktrackingScope Backtrack(*this);
+  
+  if (!Tok.is(tok::l_brace)) {
+    // Statement bodies always start with a '{'. If there is no '{', we can't be at the statement body.
+    return false;
+  }
+  
+  skipSingle();
+  
+  if (Tok.is(tok::eof)) {
+    // There's nothing else in the source file that could be the statement body, so this must be it.
+    return true;
+  }
+  
+  if (Tok.is(tok::semi)) {
+    // We can't have a semicolon between the condition and the statement body, so this must be the statement body.
+    return true;
+  }
+  
+  if (Tok.is(tok::kw_else)) {
+    // If the current token is an `else` keyword, this must be the statement body of an `if` statement since conditions can't be followed by `else`.
+    return true;
+  }
```

Comment:
I think you may be able to write lines 1847-1857 (though not sure how you would fit the comments) as:
```cpp
if (Tok.is(tok::eof) || Tok.is(tok::semi) || Tok.is(tok::kw_else)) { 
  return true;
}
```
Also do these occur in a specific order for a reason? On 1866 you have something else that will return true, but in between here and there you have something that will return false. If order does not matter then it may be cleaner to keep all the `return true` cases together.